traefik/templates/prometheusrules.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) .Values.metrics.prometheusRule.enabled }}
```

Comment:
I'm not too familiar with `.Capabilities`, but wouldn't this require a client connection to k8s to validate the API Group `monitoring.coreos.com/v1`? Wouldn't this break on things like `helm template` if it can't access a cluster?

Reply:
That is correct.  But this way we guarantee that the chart does not fail if the user has no prometheus because the CRDs do not exist

Reply:
In fact it's counterintuitive. This is an opt-in feature (value should be explicitly set to `true`), so user knows what they are doing. If CRDs aren't available in cluster helm's failure is desirable outcome. Otherwise helm will silently skip the resource, and user will end up with non-working solution.